### ISSUE TITLE
Cannot install in MacOSX

### ISSUE THREAD

**Issue by ppapad01** [kul-forbes/nmpc-codegen#2]
I'm trying to install `nmpc-codegen` on a MacOSX, High Sierra 10.13.4 using cmake, and this is what I get:

```
$ sudo cmake -H. -Bbuild
-- Configuring done
CMake Error at casadi/CMakeLists.txt:7 (add_library):
  Cannot find source file:

    cost_function_derivative_combined.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx


CMake Error at casadi/CMakeLists.txt:7 (add_library):
  No SOURCES given to target: CASADI_lib


-- Build files have been written to: /Users/myusername/Documents/GitHub/nmpc-codegen/build
```

It would be useful to have some installation instructions.

**Comment by alphaville** [comment]
@ppapad01 @Zilleplus I can reproduce the error on Ubuntu 14.04. It seems that `cost_function_derivative_combined.c` is missing.

**Comment by Zilleplus** [comment]
yes, well if two people get this issue i should clearly document it better. I will put a warning on the front page. 

Short story: ignore Cmake in the repo, just run the Matlab/Python scripts, it will call Cmake for you. 

Long story: If you  read the install instructions you will never see any mention of manually running Cmake in the repo. This is only necessary if you want to run the tests on the C-code. (so for internal developers) If you run: https://github.com/kul-forbes/nmpc-codegen/blob/master/generate_test_files.py it will generate the files needed to test, feel free to do so if you want. Then you should be able to run the tests.

Extra: **If you use a mac and Matlab and get the error "cannot find cmake command" or something like that.** It's a known bug, haven't found a nice automatic fix but after Matlab starts make sure the path in Matlab is the same as your system Path:

echo $PATH in a terminal outside Matlab must be the same as getenv('PATH') 

If it isn't equal, set the Matlab path equal to the $PATH variable (copy past it)
setenv('PATH','very long boring path variable .....');

**Comment by Zilleplus** [comment]
"It would be useful to have some installation instructions." -> front page of the repo, should say "install instructions" click on python or Matlab

**Comment by alphaville** [comment]
@Zilleplus Users need to install casadi with python3, right? I believe travis CI would help sort out most issues related to the installation of nmpc-codegen.

**Comment by Zilleplus** [comment]
Maybe, my knowledge of Travis is limited. But i was planning to package it, and make it available on pip. Then pip will solve the dependencies.  I expect that to happen this or next week. I will flag that as version 1.0.0

**Comment by alphaville** [comment]
If you're interested you could use [this .travis.yml configuration file](https://github.com/kul-forbes/scs/blob/master/.travis.yml) as an example. Making it available on `pip` would be fantastic!

**Comment by Zilleplus** [comment]
Right now i am constructing a panoc library. The same panoc code from this repo, but no nmpc stuff. More info on this later this week. But it should also work on Linux,Windows and mac. Using SWIG to interface with different programming languages.

**Comment by Zilleplus** [comment]
@ppapad01 @alphaville  I am closing the issue, i assume the problem is resolved? I added a error message in cmake, "Run generate_test_files.py first, with python3 and library casadi installed". To avoid this in the future.
